tests/h/search/client_test.py

```diff
@@ -31,75 +44,139 @@ def test_conn_is_read_only(self):
+        [(get_client, ES_PATH),
+         (get_client_old, ES1_PATH)])
+    def test_sets_aws_auth(self, settings, patch_path, patched_elasticsearch, patched_aws_auth, get_client):
         settings.update({'es.aws.access_key_id': 'foo', 'es.aws.secret_access_key': 'bar', 'es.aws.region': 'baz'})
         get_client(settings)
 
-        _, kwargs = patched_client.call_args
+        _, kwargs = patched_elasticsearch.call_args
         assert kwargs['http_auth'] == patched_aws_auth.return_value
 
-    def test_sets_connection_class_for_aws_auth(self, settings, patched_client):
+    @pytest.mark.parametrize('get_client,elasticsearch,patch_path',
+        [(get_client, elasticsearch, ES_PATH), (get_client_old, elasticsearch1, ES1_PATH)])
+    def test_sets_connection_class_for_aws_auth(self, settings, patch_path, patched_elasticsearch, get_client, elasticsearch):
         settings.update({'es.aws.access_key_id': 'foo', 'es.aws.secret_access_key': 'bar', 'es.aws.region': 'baz'})
         get_client(settings)
 
-        _, kwargs = patched_client.call_args
-        assert kwargs['connection_class'] == RequestsHttpConnection
+        _, kwargs = patched_elasticsearch.call_args
+        assert kwargs['connection_class'] == elasticsearch.RequestsHttpConnection
 
-    @pytest.mark.parametrize('aws_settings', [
-        {'es.aws.access_key_id': 'foo'},
-        {'es.aws.secret_access_key': 'foo'},
-        {'es.aws.region': 'foo'},
-        {'es.aws.access_key_id': 'foo', 'es.aws.secret_access_key': 'bar'},
-        {'es.aws.access_key_id': 'foo', 'es.aws.region': 'bar'},
-        {'es.aws.secret_access_key': 'foo', 'es.aws.region': 'bar'},
+    @pytest.mark.parametrize('aws_settings,get_client,patch_path', [
+        ({'es.aws.access_key_id': 'foo'}, get_client, ES_PATH),
+        ({'es.aws.secret_access_key': 'foo'}, get_client, ES_PATH),
+        ({'es.aws.region': 'foo'}, get_client, ES_PATH),
+        ({'es.aws.access_key_id': 'foo', 'es.aws.secret_access_key': 'bar'}, get_client, ES_PATH),
+        ({'es.aws.access_key_id': 'foo', 'es.aws.region': 'bar'}, get_client, ES_PATH),
+        ({'es.aws.secret_access_key': 'foo', 'es.aws.region': 'bar'}, get_client, ES_PATH),
     ])
-    def test_ignores_aws_auth_when_settings_missing(self, settings, patched_client, aws_settings):
+    def test_ignores_aws_auth_when_settings_missing(self, settings, patch_path, patched_elasticsearch, aws_settings, get_client):
         settings.update(aws_settings)
         get_client(settings)
 
-        _, kwargs = patched_client.call_args
+        _, kwargs = patched_elasticsearch.call_args
         assert 'auth' not in kwargs
 
     @pytest.fixture
     def settings(self):
         return {
             'es.host': 'search.svc',
+            'es.url': 'search.svc',
             'es.index': 'my-index',
         }
 
     @pytest.fixture
-    def patched_client(self, patch):
-        return patch('h.search.client.Client')
+    def patched_elasticsearch(self, patch, request):
```

Comment:
I changed this from mocking the custom Client to mocking the underlying call to elasticsearch.Elasticsearch that actually creates the client object as that is more compatible with both versions of the code.